src/main/java/com/minecolonies/coremod/colony/crafting/SifterRecipe.java

```diff
@@ -0,0 +1,101 @@
+package com.minecolonies.coremod.colony.crafting;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.util.JSONUtils;
+import net.minecraft.util.ResourceLocation;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents a sifting recipe, consisting of information about the mesh to be used,
+ * and a list of input items and the results that they produce (via loot table).
+ *
+ * Note that it is legal for multiple recipes to exist with the same mesh type, but
+ * they are expected to have different input items in that case.  The mesh properties
+ * will be taken from one of them at random.
+ */
+@SuppressWarnings({"MethodReturnOfConcreteClass", "ClassWithoutLogger"})
+public final class SifterRecipe

```

Comment:
This works great for the sifter, where the recipe doesn't yet have the mesh added to it, and mesh is the only pivot. Nether Gatherer recipes don't fit at all, and will be structured like the sifter was before this change. I'd rather see you focus on making that structure work well, as we'll have multiple crafters using it shortly.